============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /home/sgmdlt/Study/python-project-lvl2/.venv/bin/python
cachedir: .pytest_cache
rootdir: /home/sgmdlt/Study/python-project-lvl2, configfile: setup.cfg
plugins: cov-2.12.1
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
____________________ ERROR collecting tests/gendiff_test.py ____________________
ImportError while importing test module '/home/sgmdlt/Study/python-project-lvl2/tests/gendiff_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.venv/lib/python3.8/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
.venv/lib/python3.8/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
.venv/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
tests/gendiff_test.py:3: in <module>
    from tests.fixtures.nested.jsondiff import RESULT as expected
tests/fixtures/nested/jsondiff.py:1: in <module>
    from tests.gendiff_test import RESULT
.venv/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
tests/gendiff_test.py:3: in <module>
    from tests.fixtures.nested.jsondiff import RESULT as expected
E   ImportError: cannot import name 'RESULT' from partially initialized module 'tests.fixtures.nested.jsondiff' (most likely due to a circular import) (/home/sgmdlt/Study/python-project-lvl2/tests/fixtures/nested/jsondiff.py)
=========================== short test summary info ============================
ERROR tests/gendiff_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.17s ===============================
